src/main/java/graphql/analysis/QueryTransformer.java

```diff
@@ -36,20 +36,22 @@ public class QueryTransformer {
     private final GraphQLSchema schema;
     private final Map<String, FragmentDefinition> fragmentsByName;
     private final Map<String, Object> variables;
-
     private final GraphQLCompositeType rootParentType;
+    private final QueryTraversalOptions options;
```

Comment:
QueryTransformer also uses the NodeTracking code and hence this can apply here as well - why not